kong/plugins/http-log/handler.lua

```diff
@@ -149,40 +152,14 @@ function HttpLogHandler:log(conf)
     end
   end
 
-  local entry = cjson.encode(kong.log.serialize())
-
-  local queue_id = get_queue_id(conf)
-  local q = queues[queue_id]
-  if not q then
-    -- batch_max_size <==> conf.queue_size
-    local batch_max_size = conf.queue_size or 1
-    local process = function(entries)
-      local payload = batch_max_size == 1
-                      and entries[1]
-                      or  json_array_concat(entries)
-      return send_payload(self, conf, payload)
-    end
-
-    local opts = {
-      retry_count    = conf.retry_count,
-      flush_timeout  = conf.flush_timeout,
-      batch_max_size = batch_max_size,
-      process_delay  = 0,
-    }
-
-    local err
-    q, err = BatchQueue.new("http-log", process, opts)
-    if not q then
-      kong.log.err("could not create queue: ", err)
-      return
-    end
-    queues[queue_id] = q
-  end
+  local queue = Queue.get(
+    (conf.queue and conf.queue.name) or conf.__key__,
+    function(q, entries) return send_entries(q.conf, entries) end,
```

Comment:
You may want to define a function rather than generate a closure for each queue.

Reply:
This was a leftover from previous iterations, I changed the signature of send_entries so that it can be used as the handler function.